astropy/io/ascii/ui.py

```diff
@@ -661,11 +664,25 @@ def write(table, output=None,  format=None, Writer=None, fast_writer=True, **kwa
     Writer : ``Writer``
         Writer class (DEPRECATED) (default= :class:`Basic`)
     """
+
     if output is None:
         output = sys.stdout
 
     table = Table(table, names=kwargs.get('names'))
 
+    #check which key not match in names dictionary
+    if kwargs.get('formats') and kwargs.get('names'):
+        for key in kwargs['formats']:
+            if not key in kwargs['names']:
+                _diffkeyArray.append(key)
+
+    #if _diffkeyArray is not empty then it's warn
+    if len(_diffkeyArray) != 0:
+        warnings.warn(
+            '%r keys in the formats dictionary not to match the keys in names dictionary'
+            %_diffkeyArray, AstropyDeprecationWarning)
```

Comment:
Here you would use just `AstropyWarning`--this is not a deprecation.  Also use `.format()` instead of old-style `%` formatting (this is still inconsistent in some areas but new style formatting should be used for new code)